File: src/main/java/com/nkming/powermenu/PreferenceActivity.java
/*
 * PreferenceActivity.java
 *
 * Author: Ming Tsang
 * Copyright (c) 2015 Ming Tsang
 * Refer to LICENSE for details
 */

package com.nkming.powermenu;

import android.app.Fragment;
import android.content.Context;
import android.os.Bundle;
import android.preference.PreferenceManager;
import android.support.v7.app.ActionBarActivity;

public class PreferenceActivity extends ActionBarActivity
{
	@Override
	public void onCreate(Bundle savedInstanceState)
	{
		super.onCreate(savedInstanceState);
		PreferenceManager.setDefaultValues(this, getString(R.string.pref_file),
				Context.MODE_PRIVATE, R.xml.preference, false);

		setContentView(R.layout.activity_main);
		if (savedInstanceState == null)
		{
			Fragment f = PreferenceFragment.create();
			getFragmentManager().beginTransaction().add(R.id.container, f)
					.commit();
		}

		PersistentViewHelper.startIfNecessary(this);
		SystemOverrideService.startIfNecessary(this);
	}
}

File: src/main/kotlin/com/nkming/powermenu/PreferenceActivity.kt
package com.nkming.powermenu

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity

class PreferenceActivity : AppCompatActivity()
{
	override fun onCreate(savedInstanceState: Bundle?)
	{
		_themeAdapter.onCreate(savedInstanceState)
		super.onCreate(savedInstanceState)

		setContentView(R.layout.activity_main)
		if (savedInstanceState == null)
		{
			val f = PreferenceFragment.create()
			fragmentManager.beginTransaction().add(R.id.container, f).commit()
		}

		PersistentViewHelper.startIfNecessary(this)
		SystemOverrideService.startIfNecessary(this)
	}

	override fun onResume()
	{
		super.onResume()
		_themeAdapter.onResume()
	}

	override fun onPause()
	{
		super.onPause()
		_themeAdapter.onPause()
	}

	override fun onDestroy()
	{
		super.onDestroy()
		_themeAdapter.onDestroy()
	}

	private val _themeAdapter by lazy{ActivityThemeAdapter(this,
			R.style.AppThemePreference_Dark, R.style.AppThemePreference_Light)}
}
